pkg/pubsub/rabbitmq/rabbitmq.go

```diff
@@ -0,0 +1,75 @@
+package rabbitmq
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/common"
+	"github.com/open-policy-agent/gatekeeper/pkg/util"
+	"github.com/streadway/amqp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const Name = "rabbitmq"
+
+type RabbitMQ struct {
+	conn            *amqp.Connection
+	ch              *amqp.Channel
+	name            string
+	batchingEnabled bool
+}
+
+func (r *RabbitMQ) Publish(data interface{}, topic string) error {
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		return fmt.Errorf("error marshaling data: %w", err)
+	}
+
+	err = r.ch.Publish(
+		topic, // exchange
+		"",    // routing key
+		false, // mandatory
+		false, // immediate
+		amqp.Publishing{
+			ContentType: "text/plain",
+			Body:        jsonData,
+		})
+	if err != nil {
+		return fmt.Errorf("error publishing message to rabbitmq: %w", err)
+	}
+
+	return nil
+}
+
+func (r *RabbitMQ) PublishBatch(data interface{}, topic string) error {
+	// TODO: figure out a way to create a batch here and once the batch is full batch publish
+	return nil
+}
+
+func (r *RabbitMQ) NewClient(ctx context.Context, k8sClient kubeClient.Client) error {
+	var err error
+	config := corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: Name, Namespace: util.GetNamespace()}}
```

Comment:
Here we have a choice of defining our own CRD, but I feel configMap is more suitable because I think the user should be able to pass all the configurable options through data and configure the respective client accordingly.

Reply:
What is the story for secrets?